pkg/plugin/framer.go

```diff
@@ -52,3 +55,70 @@ func ConvertStatsV2ResponseToFrame(frameName string, stats sentry.StatsV2Respons
 	}
 	return frame, nil
 }
+
+type SentryEventsStatsSet struct {
+	Name string
+	Data []interface{}
+}
+
+func ConvertEventStatsSetToTimestampField(set SentryEventsStatsSet) (*data.Field){
+	field := data.NewFieldFromFieldType(data.FieldTypeTime, len(set.Data))
+	field.Name = "Timestamp"
+	for index, value := range set.Data {
+		row := value.([]interface{})
+		field.Set(index, time.Unix(int64(row[0].(float64)), 0))
+	}
+	return field
+}
+
+func ConvertEventStatsSetToField(set SentryEventsStatsSet) (*data.Field){
+	field := data.NewFieldFromFieldType(data.FieldTypeNullableFloat64, len(set.Data))
+	field.Name = set.Name
+	for index, value := range set.Data {
+		row := value.([]interface{})
+		rawCount := row[1].(([]interface{}))[0].(map[string]interface{})["count"]
+		count, ok := rawCount.(float64)
+		if ok {
+			field.Set(index, &count)
+		}
+	}
+	return field
+}
+
+func ConvertEventsStatsResponseToFrame(frameName string, eventsStats sentry.SentryEventsStats) (*data.Frame, error) {
+	sets := ExtractDataSets("", eventsStats)
+	frame := data.NewFrameOfFieldTypes(frameName, 0)
+
+	for index, set := range sets {
+		if index == 0 {
+			frame.Fields = append(frame.Fields, ConvertEventStatsSetToTimestampField(set))
+		}
+		frame.Fields = append(frame.Fields, ConvertEventStatsSetToField(set))
+	}
+	return frame, nil
+}
+
+func ExtractDataSets(namePrefix string, rawData map[string]interface{}) ([]SentryEventsStatsSet) {
```

Comment:
The recursive pattern used in this function is a smart move, but please consider edge cases where unexpected data types might cause `panic`s and handle those errors appropriately.

Reply:
Every failed cast should return an error instead of panic now 👍